website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -0,0 +1,257 @@
+
+Example warning alert: [ClusterNotUpgradeable](https://github.com/openshift/cluster-version-operator/blob/513a2fc/install/0000_90_cluster-version-operator_02_servicemonitor.yaml#L68-L76)
+
+```yaml
+- alert: ClusterNotUpgradeable
+  annotations:
+    summary: One or more cluster operators have been blocking minor version cluster upgrades for at least an hour.
+    description: In most cases, you will still be able to apply patch releases.
+      Reason {{ "{{ with $cluster_operator_conditions := \"cluster_operator_conditions\" | query}}{{range $value := .}}{{if and (eq (label \"name\" $value) \"version\") (eq (label \"condition\" $value) \"Upgradeable\") (eq (label \"endpoint\" $value) \"metrics\") (eq (value $value) 0.0) (ne (len (label \"reason\" $value)) 0) }}{{label \"reason\" $value}}.{{end}}{{end}}{{end}}"}}
+      For more information refer to 'oc adm upgrade'{{ "{{ with $console_url := \"console_url\" | query }}{{ if ne (len (label \"url\" (first $console_url ) ) ) 0}} or {{ label \"url\" (first $console_url ) }}/settings/cluster/{{ end }}{{ end }}" }}.
+    expr: |
+      max by (name, condition, endpoint) (cluster_operator_conditions{name="version", condition="Upgradeable", endpoint="metrics"} == 0)
+    for: 60m
+    labels:
+      severity: warning
+```
+
+This alert fires if one or more operators have not reported their `Upgradeable` condition as true in more than an hour.  
+The alert has a clear name and informative summary and description annotations.  
+The timeline is appropriate for allowing the operator a chance to resolve the issue automatically, avoiding the need to alert an administrator.
+
+##### Info Alerts
+
+Info level alerts represent situations an administrator should be aware of, but they don't necessarily require any action.  
+Use these sparingly, and consider instead reporting this information via Kubernetes events.
+
+Example info alert: [MultipleContainersOOMKilled](https://github.com/openshift/cluster-monitoring-operator/blob/79cdf68/assets/cluster-monitoring-operator/prometheus-rule.yaml#L326-L338)
+
+```yaml
+- alert: MultipleContainersOOMKilled
+  annotations:
+    description: Multiple containers were out of memory killed within the past
+      15 minutes. There are many potential causes of OOM errors, however issues
+      on a specific node or containers breaching their limits is common.
+      summary: Containers are being killed due to OOM
+  expr: sum(max by(namespace, container, pod) (increase(kube_pod_container_status_restarts_total[12m]))
+    and max by(namespace, container, pod) (kube_pod_container_status_last_terminated_reason{reason="OOMKilled"}) == 1) > 5
+  for: 15m
+  labels:
+    namespace: kube-system
+    severity: info
+```
+
+This alert fires if multiple containers have been terminated due to out of memory conditions in the last 15 minutes.  
+This is something the administrator should be aware of, but may not require immediate action.
+
+### Alerts, Metrics and Recording Rules Tests
```

Comment:
consider mentioning this section at the end of the Metrics Guidelines, as a remainder to include tests.

Reply:
Added comments to metrics and recording rules sections